src/platform/testing/e2e/cypress/support/commands/keyboardNavigation.js

```diff
@@ -95,3 +118,124 @@ Cypress.Commands.add(
+ */
+Cypress.Commands.add('typeInIfDataExists', (selector, text) => {
+  if (text) {
+    cy.tabToElement(selector);
+    cy.typeInFocused(text);
+  }
+});
+
+/**
+ * @callback setCheckboxFromData
+ * @param {String} selector - The CSS element selector
+ * @param {Boolean} isChecked=false - checkbox state
+ */
+/**
+ * This command will find and check a checkbox if the data is true
+ * @param {String} name - Name of the new command
+ * @param {setCheckboxFromData} callbackFunction - The callback that handles focusing & checking or unchecking a checkbox based on the checkbox state
+ */
+Cypress.Commands.add('setCheckboxFromData', (selector, isChecked = false) => {
+  cy.tabToElement(selector).then($el => {
+    if ($el[0].checked !== isChecked) {
+      cy.realPress('Space');
+    }
+  });
+});
+
+/**
+ * FullName
+ * @typedef {Object}
+ * @property {String} prefix - Prefix
+ * @property {String} first - First name
+ * @property {String} middle - Middle name
+ * @property {String} last - Last name
+ * @property {String} suffix - Suffix
+ */
+/**
+ * @callback typeInFullName
+ * @param {String} fieldName - The group of elements ID prefix
+ * @param {FullName} data - Full name object
+ */
+/**
+ * This command will check if the data portion of the value is truthy, before tabbing to the name part. If found, it will then select or type in the name part
+ * @param {String} name - Name of the new command
+ * @param {typeInFullName} callbackFunction - The callback that handles moving the focus to an element and then select or type in the text
+ */
+Cypress.Commands.add('typeInFullName', (fieldName, data) => {
+  const name = removeLeadingHash(fieldName);
+  if (data.prefix) {
+    cy.tabToElement(`#${name}prefix`);
+    cy.chooseSelectOptionByTyping(data.prefix);
+  }
+  cy.typeInIfDataExists(`#${name}first`, data.first);
+  cy.typeInIfDataExists(`#${name}middle`, data.middle);
+  cy.typeInIfDataExists(`#${name}last`, data.last);
+  if (data.suffix) {
+    cy.tabToElement(`#${name}suffix`);
+    cy.chooseSelectOptionByTyping(data.suffix);
+  }
+});
+
+/**
+ * @callback typeInDate
+ * @param {String} fieldName - The group of elements ID prefix
+ * @param {String} dateString - Date string in "YYYY-MM-DD" format
+ */
+/**
+ * This command will select the date in a month & day select and then type in the year into a text input
+ * @param {String} name - Name of the new command
+ * @param {typeInDate} callbackFunction - The callback that handles making the month & day selections and typing in the year
+ */
+Cypress.Commands.add('typeInDate', (fieldName, dateString) => {
+  // Remove leading zeros
+  const date = dateString.split('-').map(v => parseInt(v, 10).toString());
+  const name = removeLeadingHash(fieldName);
+  cy.tabToElement(`#${name}Month`);
+  cy.chooseSelectOptionUsingValue(date[1]);
+  cy.tabToElement(`#${name}Day`);
+  cy.chooseSelectOptionUsingValue(date[2]);
+  cy.tabToElement(`input[name="${name}Year"]`);
+  cy.typeInFocused(date[0]);
+});
```

Comment:
Commands that may still be needed include:
- Upload a file
- Enter address (w/military base checkbox)